### PR TITLE
add support for HTTP Basic Auth and fix  HTTP headers 

### PR DESCRIPTION
**Pull request by lasse-aagren**
simple HTTP Basic auth support using Faraday's basic auth middleware and fixing code where http headers were unconditionally overwritten, thus removing added auth.

### CODE REVIEW

**Comment by lasse-aagren** [comment]
will make a pull request on active_fedora as well, if this request is accepted

https://github.com/lasse-aagren/active_fedora/commit/d3aef611d24d5a22891c5117e6dc60a7530425c5

**Comment by jcoyne** [comment]
This seems to have broken the CI build. Please fix it.  Also, please add some tests for your changes.

File: lib/ldp/client/methods.rb [hunk 1]
```diff
@@ -8,11 +8,24 @@ module Ldp::Client::Methods
   def initialize_http_client *http_client
```
Comment (cbeer) [inline lib/ldp/client/methods.rb]: I'm a little scared of this signature (and, yes, it's at least partly my fault.) 

I'm not a fan of the 3-arg option you're proposing. Could we just yield the http client in `Ldp::Client#initialize` and let our downstream consumers do whatever?

Comment (lasse-aagren) [inline lib/ldp/client/methods.rb]: I'm fine with either - but we still need LDP not to unconditionally overwrite HTTP headers.

**Comment by lasse-aagren** [comment]
@jcoyne tests fixed/added

I don't the jruby  build failure is caused by me?

>An error occurred while installing slop (4.0.0), and Bundler cannot continue.
>Make sure that `gem install slop -v '4.0.0'` succeeds before bundling.

**Comment by lasse-aagren** [comment]
@cbeer: a solution where the downstream consumer handles basic_auth:

https://github.com/lasse-aagren/ldp/commit/f687abfcf86a741d4c9731081e6893669491b68a

and an example of how the downstream consumer can use it:

https://github.com/lasse-aagren/active_fedora/commit/e8e69f58f3a856c4a1d0813b118d29bb78150910

Let me know if I should create a pull request on this branch instead.

**Comment by jcoyne** [comment]
@lasse-aagren I like that new solution!  Please create a PR on that branch.

File: lib/ldp/resource/rdf_source.rb [hunk 2]
```diff
@@ -17,7 +17,9 @@ def initialize client, subject, graph_or_response = nil, base_path = ''
 
     def create
       super do |req|
-        req.headers = { "Content-Type" => "text/turtle" }
+        if req.headers["Content-Type"] != "text/turtle"
```
Comment (jcoyne) [inline lib/ldp/resource/rdf_source.rb]: Is this really necessary?  Seems like a extra check for no benefit.

Comment (lasse-aagren) [inline lib/ldp/resource/rdf_source.rb]: so just remove the if-conditional and just set:

req.headers["Content-Type"] = "text/turtle"

right? can't remember why I made the conditional :)

**Comment by lasse-aagren** [comment]
@jcoyne New PR created:

https://github.com/cbeer/ldp/pull/38

**Pull request closed without merging**